containers/unit_tests/TestDualView.hpp

```diff
@@ -60,6 +60,48 @@ struct test_dualview_alloc {
   }
 };
 
+template <typename Scalar, class Device>
+struct test_dualview_copy_construction_and_assignment {
+  using scalar_type     = Scalar;
+  using execution_space = Device;
+
+  template <typename ViewType>
+  void run_me() {
+    const unsigned int n = 10;
+    const unsigned int m = 5;
+
+    ViewType a("A", n, m);
+
+    // Copy construction
+    ViewType b(a);
+
+    // Copy assignment
+    ViewType c = a;
```

Comment:
This should have worked before already. Can you modify this or add another case so that the View types are compatible but not equal?

Reply:
Hi @masterleinad. OK, indeed. I've just updated it so that the views are compatible but not of the same type.

Reply:
yeah that was the thing i was saying in my suggestion for checking assignment of a compatible view